.circleci/config.yml

```diff
@@ -1,148 +1,552 @@
+          command: bundle exec appraisal install
+      - run:
+          name: Compute bundle checksum
+          command: |
+            cat Gemfile.lock gemfiles/*.gemfile.lock > .circleci/bundle_checksum
+      - save_cache:
+          key: '{{ .Environment.CIRCLE_CACHE_VERSION }}-bundled-repo-2.0-{{ .Environment.CIRCLE_SHA1 }}'
+          paths:
+            - /app
+      - save_cache:
+          key: '{{ .Environment.CIRCLE_CACHE_VERSION }}-bundle-2.0-{{ checksum ".circleci/bundle_checksum" }}'
+          paths:
+            - /usr/local/bundle
+  test-2.0:
+    working_directory: /app
+    shell: /bin/bash --login
+    docker:
+      - image: datadog/docker-library:ddtrace_rb_2_0_0
+        environment:
+          - TEST_DATADOG_INTEGRATION: 1
+      - image: postgres:9.6
+        environment:
+          - POSTGRES_PASSWORD=postgres
+          - POSTGRES_USER=postgres
+          - POSTGRES_DB=postgres
+      - image: mysql:5.6
+        environment:
+          - MYSQL_ROOT_PASSWORD=root
+          - MYSQL_PASSWORD=mysql
+          - MYSQL_USER=mysql
+      - image: elasticsearch:2.4
+      - image: redis:3.0
+      - image: mongo:3.5
+      - image: memcached:1.5-alpine
+      - image: datadog/docker-dd-agent
+        environment:
+          - DD_APM_ENABLED=true
+          - DD_BIND_HOST=0.0.0.0
+          - DD_API_KEY=invalid_key_but_this_is_fine
+    steps:
+      - restore_cache:
+          keys:
+            - '{{ .Environment.CIRCLE_CACHE_VERSION }}-bundled-repo-2.0-{{ .Environment.CIRCLE_SHA1 }}'
+      - restore_cache:
+          keys:
+            - '{{ .Environment.CIRCLE_CACHE_VERSION }}-bundle-2.0-{{ checksum ".circleci/bundle_checksum" }}'
+      - run:
+          name: Run tests
+          command: bundle exec rake ci
+  checkout-2.1:
+    working_directory: /app
+    shell: /bin/bash --login
+    docker:
+      - image: datadog/docker-library:ddtrace_rb_2_1_10
+    steps:
+      - checkout
+      - save_cache:
+          key: '{{ .Environment.CIRCLE_CACHE_VERSION }}-repo-2.1-{{ .Environment.CIRCLE_SHA1 }}'
+          paths:
+            - /app
+  build-2.1:
+    working_directory: /app
+    shell: /bin/bash --login
+    docker:
+      - image: datadog/docker-library:ddtrace_rb_2_1_10
+    steps:
+      - restore_cache:
+          keys:
+            - '{{ .Environment.CIRCLE_CACHE_VERSION }}-bundled-repo-2.1-{{ .Environment.CIRCLE_SHA1 }}'
+            - '{{ .Environment.CIRCLE_CACHE_VERSION }}-repo-2.1-{{ .Environment.CIRCLE_SHA1 }}'
+      - run:
+          name: Initialize bundle cache key
+          command: |
+            touch .circleci/bundle_checksum
+      - restore_cache:
+          keys:
+            - '{{ .Environment.CIRCLE_CACHE_VERSION }}-bundle-2.1-{{ checksum ".circleci/bundle_checksum" }}'
+      - run:
+          name: Install gem dependencies
+          command: bundle install
+      - run:
+          name: Delint with Rubocop
+          command: bundle exec rake rubocop
```

Comment:
Right now it does fail separately from the `test` jobs. As it's configured, it's meant to "fail fast", even before all the dependencies are installed, so you know when there's a Rubocop issue before waiting on a long build.

We could make a case for putting Rubocop in its own task after `build`, since it makes for a more aesthetically organized build, but we'll end up installing gems and taking longer just to fail as a trade off.

Reply:
I meant we could make configuration that does Rubocop and whatever else static analysis  as separate build task. I don't think `rubocop` needs to run with every build for every configuration we have. One run should be enough for the whole change.

The end result would be that we'll have another check reported by GitHub here.
![image](https://user-images.githubusercontent.com/812891/40786527-83e642dc-64eb-11e8-866d-9b65a893b14d.png)

The benefit of that is since Rubocop often only breaks on trivial things like whitespace in the wrong place, by extracting it as a separate check that runs in parallel. We would still have fast Rubocop feedback, but the tests that are most important would still run irregardless of rubocop result. 

At least my workflow would benefit from that setup since I'd get feedback about test breaking sooner. Right now I need to submit perfectly formatted code everytime.